packages/dev/core/src/Engines/abstractEngine.ts

```diff
@@ -889,16 +889,59 @@ export abstract class AbstractEngine {
     public abstract get performanceMonitor(): PerformanceMonitor;
 
     /** @internal */
-    public _boundRenderFunction: any = () => this._renderLoop();
+    public _boundRenderFunction: any = (timestamp: number) => this._renderLoop(timestamp);
 
-    /** @internal */
-    public _renderLoop(): void {
-        // Reset the frame handler before rendering a frame to determine if a new frame has been queued.
+    protected _maxFPS: number | undefined;
+    protected _minFrameTime: number;
+    protected _lastFrameTime: number = 0;
+
+    /**
+     * Skip frame rendering but keep the frame heartbeat (begin/end frame).
+     * This is useful if you need all the plumbing but not the rendering work.
+     * (for instance when capturing a screenshot where you do not want to mix rendering to the screen and to the screenshot)
+     */
+    public skipFrameRender = false;
+
+    /** Gets or sets max frame per second allowed */

```

Comment:
nit: maybe add in the doc that if no max has been defined, "undefined" will be returned by the getter?